src/h5web/visualizations/heatmap/utils.ts

```diff
@@ -1,26 +1,35 @@
-import React, { useCallback, useRef, useEffect } from 'react';
 import { extent } from 'd3-array';
 import { rgb } from 'd3-color';
-import { scaleLinear, scaleSequential } from 'd3-scale';
-import { Vector3 } from 'three';
-import { ReactThreeFiber, PointerEvent, useThree } from 'react-three-fiber';
-import { clamp, range } from 'lodash-es';
+import {
+  scaleSequential,
+  ScaleLinear,
+  ScaleSymLog,
+  scaleLinear,
+  scaleSymlog,
+} from 'd3-scale';
+import { range } from 'lodash-es';
 import { D3Interpolator } from './interpolators';
 
-const ZOOM_FACTOR = 0.95;
+export type ColorScale =
+  | ScaleLinear<number, number>
+  | ScaleSymLog<number, number>;
 
 export function computeTextureData(
   values: number[],
-  domain: [number, number] | undefined,
+  colorScale: ColorScale | undefined,
   interpolator: D3Interpolator
 ): Uint8Array | undefined {
-  if (domain === undefined) {
+  if (colorScale === undefined) {
     return undefined;
   }
-  const scale = scaleSequential(interpolator).domain(domain);
+
+  // Map colors to the output of colorScale
+  const scale = scaleSequential(interpolator).domain(
+    colorScale.range() as [number, number]
+  );
   // Compute RGB color array for each datapoint `[[<r>, <g>, <b>], [<r>, <g>, <b>], ...]`
   const colors = values.map(val => {
-    const { r, g, b } = rgb(scale(val)); // `colorMap` returns CSS RGB strings
+    const { r, g, b } = rgb(scale(colorScale(val))); // `scale` returns CSS RGB strings
```

Comment:
The scaleSequential must map the colors to the output of the colorScale (which are either a linear extrapolation of the data or a log extrapolation).
In short, we have: `data` ---`colorScale`----> `scaledData` ---`scaleSequential`---> `colors`

Reply:
Do you think there's a better way to name the scales ?

Reply:
I thought about `preScale` but that's ugly. I like `colorScale` as it means that the data is scaled either linearly or logarithmically to the colors. It just happens that in the implementation, we need a `scaleSequential`.
This being said, I concur that it may be misleading and if you have a suggestion, go ahead !

Reply:
Will be changed in the next PR